harness/determined/pytorch/_pytorch_context.py

```diff
@@ -477,12 +531,12 @@ def clip_grads(params):
         if self._should_communicate_and_update():
             # Communication needs to be synchronized so that is completed
             # before we apply gradient clipping and `step()`.
```

Comment:
We do not support multiple losses with AMP.

It seems to me that we were very indecisive on how much we supported apex amp because we have e.g. the `num_losses` parameter but we also have checks against calling `context.backward()` multiple times in a single `train_batch`.  We also wrap the `optimizer.synchronize()` call in a `self._should_communicate_and_update()` check but we have checks to guarantee that amp is not used when aggregation frequency >1.

I am not sure why it was written that way anymore (did we think we could eventually solve those problems?) but this code could stand to get a lot clearer by just explaining those assumptions in the places where we depend on them.

I'm pretty happy with the comment as you have it right now.